pkg/cloud/azure/errors.go

```diff
@@ -27,14 +29,16 @@ func ResourceNotFound(err error) bool {
 	if derr, ok := err.(autorest.DetailedError); ok && derr.StatusCode == 404 {
 		return true
 	}
-	return false
+	return azureerrors.IsNotFoundErr(err)
```

Comment:
Is this now what we are trying to achieve with the previous if statement? Is one redundant?

Reply:
Unfortunately not, no. `azureerrors` expects errors to be generated by the new SDK. We need to check both.